pkg/dependency/parser/java/pom/pom.go

```diff
@@ -113,12 +115,29 @@ func (p pom) licenses() []string {
 	})
 }
 
-func (p pom) repositories() []string {
+func (p pom) repositories(servers []Server) []string {
 	var urls []string
 	for _, rep := range p.content.Repositories.Repository {
-		if rep.Releases.Enabled != "false" {
-			urls = append(urls, rep.URL)
+		// Add only enabled repositories
+		if rep.Releases.Enabled == "false" && rep.Snapshots.Enabled == "false" {
+			continue
+		}
+
+		repoURL, err := url.Parse(rep.URL)
+		if err != nil {
+			log.Logger.Debugf("Unable to parse remote repository url: %s", err)
+			continue
+		}
+
+		for _, server := range servers {
+			if rep.ID == server.ID && server.Username != "" && server.Password != "" {
+				repoURL.User = url.UserPassword(server.Username, server.Password)
+				break
+			}
 		}
```

Comment:
Can you please add a comment? Does it try to merge global settings into repositories in pom.xml?

Reply:
Sure

Reply:
@knqyf263 I have added comments